scripts/pregenerate/__init__.py

```diff
@@ -98,5 +100,5 @@ def FindPregenerationTargets(sdk_root: str, filter: TargetFilter, runner):
                 continue
 
         for generator in generators:
-            if generator.Accept(idl):
+            if generator.Accept(idl, filter.cpp_only):
```

Comment:
cpp only seems to be essentially a filter for java. Would it not be apropriate to not have the java generator available then rather than making every generator have this flag added?

Reply:
Basically I am wondering if the way to change things is to make addition of `CodegenJavaPregenerator` conditional on this flag and not attach it to a filter and accept parameter.

Reply:
Yes, I can do that.

Reply:
Filed https://github.com/project-chip/connectedhomeip/issues/24553